dialects/mysql/src/main/kotlin/app/cash/sqldelight/dialects/mysql/MySqlType.kt

```diff
@@ -61,7 +61,7 @@ internal enum class MySqlType(override val javaType: TypeName) : DialectType {
     return CodeBlock.of(
       when (this) {
         TINY_INT, TINY_INT_BOOL, SMALL_INT, INTEGER, BIG_INT, BIT -> "$cursorName.getLong($columnIndex)"
-        DATE, TIME, DATETIME, TIMESTAMP -> "$cursorName.getObject($columnIndex)"
+        DATE, TIME, DATETIME, TIMESTAMP -> "$cursorName.getObject<$javaType>($columnIndex)"
```

Comment:
Using string interpolation means this type does not participate in imports. Each branch will need to be changed to individually call `CodeBlock.of` and then this should look like `CodeBlock.of("$cursorName.getObject<%T>($columnIndex)", javaType)`.

Same for the other file.